src/applicationcomposer/activation.ts

```diff
@@ -4,12 +4,41 @@
  */
 
 import * as vscode from 'vscode'
+import * as CloudFormation from '../shared/cloudformation/cloudformation'
 import { ApplicationComposerManager } from './webviewManager'
 import { Commands } from '../shared/vscode/commands2'
 import { ToolkitError } from '../shared/errors'
 import { AuthUtil, getChatAuthState } from '../codewhisperer/util/authUtil'
 import { telemetry } from '../shared/telemetry/telemetry'
 
+export class OpenApplicationComposerCodeLensProvider implements vscode.CodeLensProvider {
+    public async provideCodeLenses(
+        document: vscode.TextDocument,
+        token: vscode.CancellationToken
+    ): Promise<vscode.CodeLens[]> {
+        const cfnTemplate =
+            CloudFormation.isValidFilename(document.uri) && document.languageId === 'yaml'
+                ? await CloudFormation.tryLoad(document.uri)
+                : undefined
+
+        if (cfnTemplate?.template === undefined) {
+            return []
+        }
+
+        const lines = document.getText().split('\n')
+        for (let i = 0; i < lines.length; i++) {
+            if (lines[i].trim() === 'Resources:') {
```

Comment:
Nit: could we change this to `if (lines[i].startsWith('Resources:') || lines[i].startsWith('Transform:'))`?

Reply:
This should ideally cover all of the cases described [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-anatomy.html), though most templates start with Transform, Resources, or AWSTemplateFormatVersion.